crates/turbopack-ecmascript/src/analyzer/mod.rs

```diff
@@ -1336,9 +1505,54 @@ impl JsValue {
                 LogicalOperator::And => all_if_known(list, JsValue::is_truthy),
                 LogicalOperator::Or => any_if_known(list, JsValue::is_truthy),
                 LogicalOperator::NullishCoalescing => {
-                    shortcircut_if_known(list, JsValue::is_not_nullish, JsValue::is_truthy)
+                    shortcircuit_if_known(list, JsValue::is_not_nullish, JsValue::is_truthy)
                 }
             },
+            JsValue::Binary(_, box a, op, box b) => {
+                let (positive_op, negate) = op.positive_op();
+                match (positive_op, a, b) {
+                    (
+                        PositiveBinaryOperator::StrictEqual,
+                        JsValue::Constant(a),
+                        JsValue::Constant(b),
+                    ) if a.is_value_type() => Some(a == b),
+                    (
+                        PositiveBinaryOperator::StrictEqual,
+                        JsValue::Constant(a),
+                        JsValue::Constant(b),
+                    ) if a.is_value_type() => {
```

Comment:
Can this second branch ever execute? It's the same match as the first.

Reply:
yeah, the second branch can be removed. The first branch is an less verbose version of the first one